src/widgets/QDisassemblyView.cpp

```diff
@@ -989,18 +989,25 @@ void QDisassemblyView::paintEvent(QPaintEvent *) {
 			QString immed_string;
 			for (unsigned int op_idx = 0; op_idx < op_count; op_idx++) {
 				auto oper = inst.operands()[op_idx];
-				edb::address_t address = 0;
+				edb::address_t ascii_address = 0;
 				if (oper.type() == edb::Operand::TYPE_REL || oper.type() == edb::Operand::TYPE_IMMEDIATE) {
-					address = oper.relative_target();
-				} else if (oper.type() == edb::Operand::TYPE_EXPRESSION) {
-					// TODO Interact with capstone to evaluate operand's displacement
+					ascii_address = oper.relative_target();
+				} else if (
+					oper.type() == edb::Operand::TYPE_EXPRESSION &&
+					oper.expression().displacement_type == edb::Operand::DISP_PRESENT)
+				{
+					if (oper.expression().base == edb::Operand::Register::X86_REG_RIP) {
+						ascii_address += address + oper.owner()->size() + oper.expression().displacement;
+					} else if (oper.expression().displacement > 0) {
+						ascii_address = oper.expression().displacement;
```

Comment:
Won't this give wrong results for code like `mov eax, [ecx+edx*4+disp]` (interpreting it as if it was `mov eax,[disp]`)?